src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import React, { useState } from 'react';
+import cn from 'classnames';
+import './Carousel.scss';
+import { Options } from '../Options/Options';
+
+interface Props {
+  images: string[];
+  itemWidth: number;
+  frameSize: number,
+  step: number,
+  animationDuration: number,
+}
+
+type TypedOptions = [
+  string,
+  number,
+  React.Dispatch<React.SetStateAction<number>>,
+][];
+
+export const Carousel: React.FC<Props> = ({
+  images,
+  itemWidth,
+  frameSize,
+  step,
+  animationDuration,
+}) => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [isPrevBtnDisabled, setPrevBtnDisabled] = useState(true);
+  const [isNextBtnDisabled, setNextBtnDisabled] = useState(false);
+  const [imgWidth, setImgWidth] = useState(itemWidth);
+  const [imgCount, setImgCount] = useState(frameSize);
+  const [carouselStep, setCarouselStep] = useState(step);
+  const [duration, setDuration] = useState(animationDuration);
```

Comment:
```suggestion
  const [currentSlide, setCurrentSlide] = useState(0);
```
enough to keep in state currentSlide.
isPrevBtnDisabled and isNextBtnDisabled will be change based on currentSlide.

imgWidth, imgCount, carouselStep duration already in parent component state, no need to duplicate them here in state.